### PR TITLE
Section 2 - Neville Medhora

### PR DESCRIPTION
**Pull request by elliven11**

### CODE REVIEW

File: apps/01PigLatin.js [hunk 1]
```diff
@@ -1,16 +1,94 @@
 'use strict';
-
 var assert = require('assert');
 var prompt = require('prompt');
 prompt.start();
 
 
-function pigLatin(word) {
+function pigLatin(word)
```
Comment (yangc5) [inline apps/01PigLatin.js]: please finish the PigLatin logic. You have a great start!

File: apps/01PigLatin.js [hunk 2]
```diff
@@ -48,3 +127,4 @@ if (typeof describe !== 'undefined') {
     getPrompt();
 
 }
+
```
Comment (paulrenenichols) [inline apps/01PigLatin.js]: Let's try to get you caught up!

File: apps/03TicTacToe.js [hunk 3]
```diff
@@ -4,14 +4,20 @@ var assert = require('assert');
...
+    ['--', '-', '--'],
+    ['--', '-', '--']
 ];
 
 var playerTurn = 'X';
 
+
+function togglePlayer() {
+  playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+}
```
Comment (paulrenenichols) [inline apps/03TicTacToe.js:19]: I like that you put this in a function! Awesome!

File: apps/03TicTacToe.js [hunk 4]
```diff
@@ -22,31 +28,53 @@ function printBoard() {
 }
 
 function horizontalWin() {
-    // Your code here
+    if ( (board[0][0] === board[0][1] && board[0][0] === board[0][2]) || (board[1][0] === board[1][1] && board[1][0] ===board[1][2]) || (board[2][0] === board[2][1] && board[2][2] === board[2][1]) ) 
+       {return true;}
 }
```
Comment (paulrenenichols) [inline apps/03TicTacToe.js:33]: Your board tests work, and are clearly well thought out.

You could just return the value of the expression:

``` javascript
return (board[0][0] === board[0][1] && board[0][0] === board[0][2]) || 
       (board[1][0] === board[1][1] && board[1][0] === board[1][2]) || 
       (board[2][0] === board[2][1] && board[2][2] === board[2][1]);
```

File: test/02LessonTwoTest.js [hunk 5]
```diff
@@ -7,11 +7,32 @@ function whichSpecies(character) {
...
 
 }
 
 function isEven(number) {
     // should return true is number is even (divisible by 2)
+    if ( (number % 2) === 0){
+        return true;
+
+    }
+
```
Comment (paulrenenichols) [inline test/02LessonTwoTest.js:36]: I see you're making progress catching up!

You're missing something here, though. **isEven(number)** should return false when the number is not even.

File: apps/05Mastermind.js [hunk 6]
```diff
@@ -26,15 +26,64 @@ function getRandomInt(min, max) {
...
 }
 
+
+
+
+
 function mastermind(guess) {
-    // your code here
+    
+    var hint = board.push(generateHint(solution, guess));
```
Comment (paulrenenichols) [inline apps/05Mastermind.js]: Make sure that you're only generating if the game is still going.

And don't push if your board is already full!

You should generate your hint, and pass it to a function that colors it, then push that.

File: apps/05Mastermind.js [hunk 7]
```diff
@@ -26,15 +26,64 @@ function getRandomInt(min, max) {
...
+
+    solution = "abcd";
```
Comment (paulrenenichols) [inline apps/05Mastermind.js]: Need to call `generateSolution()` here.

File: apps/07jquery-tic-tac-toe/script.js [hunk 8]
```diff
@@ -1,5 +1,63 @@
...
+	//set all the click functions
+	$('[data-cell]').click(function () {
+		$(this).text(playerTurn);
+		
+		changePlayer();
+
+		//checkForWin announce
+		if (checkForWin()){
+			$('#announce-winner').text('Player ' + playerTurn + 'Wins!!! OMG OMG OMG!!');
+		}
```
Comment (paulrenenichols) [inline apps/07jquery-tic-tac-toe/script.js:52]: Great Job!

**Comment by paulrenenichols** [comment]
Where's Lesson 10??

**Pull request closed without merging**